treeherder/client/thclient/client.py

```diff
@@ -709,11 +712,21 @@ def _get_uri(self, project, endpoint, data=None, oauth_key=None,
 
         return uri
 
+    def _get_uri(self, endpoint):
+        uri = '{0}://{1}/api/{2}'.format(
+            self.protocol, self.host, endpoint)
+
+        return uri
+
     def _get_json(self, project, endpoint, timeout, **params):
         if timeout is None:
             timeout = self.timeout
 
-        uri = self._get_uri(project, endpoint)
+        if project is None:
+            uri = self._get_uri(endpoint)
+        else:
+            uri = self._get_project_uri(project, endpoint)
+
```

Comment:
I think that makes sense, but it would be better if project was a keyword argument in this case:

i.e.

    def _get_json(self, endpoint, timeout, project=None, **params):

That makes what's going on from the callee stage a little more obvious.